### PR TITLE
Convert Submission to Hibernate

### PR DESCRIPTION
**Pull request by apmasell**
JIRA: GLT-1386

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java [hunk 1]
```diff
@@ -25,91 +25,87 @@
...
-  private Date creationDate;
-  private Date submissionDate;
-  private boolean verified;
+  private String alias;
   private boolean completed;
+  private Date creationDate;
   private String description;
-  private String title;
+  @ManyToMany(targetEntity = PartitionImpl.class)
+  @JoinTable(name = "Submission_Partition_Dilution", joinColumns = { @JoinColumn(name = "dilution_dilutionId") }, inverseJoinColumns = {
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: `joinColumn` are for the submission, `inverseJoinColumn` for the partition and dilution. Please fix this for all `@JoinTable` annotations in this class.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java [hunk 2]
```diff
@@ -25,91 +25,87 @@
...
   private String accession;
-  private Date creationDate;
-  private Date submissionDate;
-  private boolean verified;
+  private String alias;
   private boolean completed;
+  private Date creationDate;
   private String description;
-  private String title;
+  @ManyToMany(targetEntity = PartitionImpl.class)
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: targetEntity = LibraryDilution.class

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java [hunk 3]
```diff
@@ -25,91 +25,87 @@
...
+  @JoinTable(name = "Submission_Partition_Dilution", joinColumns = { @JoinColumn(name = "dilution_dilutionId") }, inverseJoinColumns = {
+      @JoinColumn(name = "partition_partitionId") })
+  @MapKeyJoinColumn(name = "submission_submissionId")
+  @MapKeyClass(LibraryDilution.class)
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: Does it need this if LibraryDilution is our only implementation of Dilution?

Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: Sadly, yes.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java (hunk 1 above)
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: Fixed.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java (hunk 2 above)
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: The value of the map should be the target entity. The key (dilution) is `@MapClass`.

Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: That is super confusing, but the looking up I've done seems to not disagree!

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java [hunk 4]
```diff
@@ -25,91 +25,87 @@
...
+  private Map<Dilution, SequencerPoolPartition> dilutions;
+  @ManyToMany(targetEntity = ExperimentImpl.class)
+  @JoinTable(name = "Submission_Experiment", joinColumns = { @JoinColumn(name = "experiments_experimentId") }, inverseJoinColumns = {
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: Please also fix `@JoinTable` joinColumns annotation here, and for Submission_Sample, and for Submission_Study.

Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: Fixed.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java [hunk 5]
```diff
@@ -74,17 +74,18 @@
   @MapKeyClass(LibraryDilution.class)
   private Map<Dilution, SequencerPoolPartition> dilutions;
   @ManyToMany(targetEntity = ExperimentImpl.class)
-  @JoinTable(name = "Submission_Experiment", joinColumns = { @JoinColumn(name = "experiments_experimentId") }, inverseJoinColumns = {
-      @JoinColumn(name = "submission_submissionId") })
+  @JoinTable(name = "Submission_Experiment", joinColumns = {
+      @JoinColumn(name = "submission_submissionsubmission_submission") }, inverseJoinColumns = {
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: column name is `submission_submissionId`

Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: My brain today. :-(

**Comment by callunity** [comment]
Monday! :confetti_ball: :cloud_with_rain:

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/SubmissionManager.java [hunk 6]
```diff
@@ -34,57 +38,19 @@
...
-  public void setSubmissionEndPoint(O o);
-
-  /**
-   * Returns the submissionEndPoint of this SubmissionManager object.
-   * 
-   * @return O submissionEndPoint.
-   */
-  public O getSubmissionEndPoint();
-
+public interface SubmissionManager {
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/SubmissionManager.java]: Javadocs please

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/UserAuthMisoRequestManager.java [hunk 7]
```diff
@@ -591,9 +587,7 @@ public Study getStudyById(long studyId) throws IOException {
 
   @Override
   public Submission getSubmissionById(long submissionId) throws IOException {
-    Submission o = backingManager.getSubmissionById(submissionId);
-    if (readCheck(o)) return o;
-    else throw new AuthorizationIOException("User " + getCurrentUsername() + " cannot read Submission " + submissionId);
+    return backingManager.getSubmissionById(submissionId);
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/UserAuthMisoRequestManager.java]: The submission will include other entities that the user may not have read permissions on. This means that anything that has been included in a submission becomes readable to all users in a way.

The permissions were not set up correctly before and did not work then either. I'm not sure what the solution is. Make a ticket?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/submission/EndPoint.java [hunk 8]
```diff
@@ -31,14 +31,14 @@
...
-public interface EndPoint<T extends URI> {
+public interface EndPoint {
   /**
    * Accepts sequence Datafiles
    * 
    * @param destination
    *          String
    */
-  public void setDestination(T destination);
+  public void setDestination(URI destination);
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/submission/EndPoint.java]: If this now has to be a URI, is there any point to subclasses, or even to this being a class? Just use a URI instead?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/ERASubmissionManager.java [hunk 9]
```diff
@@ -95,7 +102,7 @@
  * @author Rob Davey
  * @since 0.0.2
  */
-public class ERASubmissionManager implements SubmissionManager<Set<Submittable<Document>>, URL, Document> {
+public class ERASubmissionManager implements SubmissionManager {
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/ERASubmissionManager.java]: GLT-1433

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/ERASubmissionManager.java [hunk 10]
```diff
@@ -494,15 +795,7 @@ public UploadReport getUploadProgress(Long submissionId) {
...
-  @Override
-  public void setTransferMethod(TransferMethod transferMethod) {
-
-  }
-
-  /*
-   * public FTPUploadReport getUploadReport(Submission submission){ return uploadReports.get(submission); }
-   */
-
+  // TODO(apmasell): These files no longer get written to disk, so this method doesn't work.
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/ERASubmissionManager.java]: Can we call that out of scope for this PR?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/UserAuthMisoRequestManager.java (hunk 7 above)
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/UserAuthMisoRequestManager.java]: GLT-1434

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/submission/EndPoint.java (hunk 8 above)
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/submission/EndPoint.java]: Fixed.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java [hunk 11]
```diff
@@ -25,91 +25,88 @@
...
-  private Date creationDate;
-  private Date submissionDate;
-  private boolean verified;
+  private String alias;
   private boolean completed;
+  private Date creationDate;
   private String description;
-  private String title;
+  @ManyToMany(targetEntity = PartitionImpl.class)
+  @JoinTable(name = "Submission_Partition_Dilution", joinColumns = { @JoinColumn(name = "partition_partitionId") }, inverseJoinColumns = {
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SubmissionImpl.java]: Fixed.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/SubmissionManager.java (hunk 6 above)
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/SubmissionManager.java]: Fixed.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/ERASubmissionManager.java (hunk 10 above)
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/ERASubmissionManager.java]: As long as it's in the scope of "before we merge into develop," in which case more tickets please :tickets: :tickets: :tickets: :tickets: :tickets: ...

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/SubmissionManager.java [hunk 12]
```diff
@@ -32,12 +32,6 @@
 import uk.ac.bbsrc.tgac.miso.core.exception.SubmissionException;
 import uk.ac.bbsrc.tgac.miso.core.service.submission.UploadReport;
 
-/**
- * An interface that defines a SubmissionManager object that can submit objects to a service defined by an endpoint
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/SubmissionManager.java]: No deleting Javadocs - update only! :neckbeard:

Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/SubmissionManager.java]: Fixed.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/ERASubmissionManager.java (hunk 10 above)
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/ERASubmissionManager.java]: ?

**Pull request merged**